integration-libs/cds/src/components/recent-searches/recent-searches.service.ts

```diff
@@ -18,11 +18,12 @@ export class RecentSearchesService {
 
   public get recentSearches$(): Observable<string[]> {
     this.addRecentSearchesListener();
-    return this.recentSearchesSource.asObservable();
+    return this.recentSearchesSource;
   }
+
   checkAvailability() {
     return interval(150).pipe(
-      concatMap((_) => of((<any>this.winRef.nativeWindow).Y_TRACKING)),
+      concatMap((_) => of((<any>this.winRef.nativeWindow)?.Y_TRACKING)),
       take(5),
       takeWhile((result: any) => !result.recentSearches),
       endWith(true)
```

Comment:
https://github.com/SAP/spartacus/pull/18400#discussion_r1479795911

Reply:
What do you think about this implementation:

```
checkAvailability() {
   return interval(150).pipe(
    concatMap(() => of((<any>this.winRef.nativeWindow)?.Y_TRACKING)),
    map(result => !!result?.recentSearches),
    take(5),
    takeWhile(val => !val, true),
    last(),
}
```

- `takeWhile(val => !val, true)`: This operator is now used to continue taking values until a `truthy` one is found, but importantly, it includes the `inclusive` parameter set to `true`. This means it will emit the last checked value (whether true or false) when it stops taking values. This change is crucial because it ensures that if `Y_TRACKING.recentSearches` is found on the first attempt, the observable will complete correctly after emitting `true`.

- `last()`: Ensures that the last emitted value is taken. This is needed to grab the last value emitted by `takeWhile` — either the first occurrence of `true` or `false` if `Y_TRACKING.recentSearches` is not found within 5 attempts.

This way `checkAvailability()` will emit only one value either `true` or `false`.

In your implementation, if recentSearches will be undefined, if you log values in subscribe you will see 5 logs of undefined and 1 true. Two problems here,  that we are not interested in this undefined values and value `true` in this case should be rather `false`